src/substrait/textplan/README.md

```diff
@@ -0,0 +1,208 @@
+# Substrait Text Plan Language Reference
+
+## Types
+Types are the core at the core of Substrait plans.  Functions consume data of a particular type and output data in the 
+same or a different type.  Optional types in Substrait are represented by appending a question mark to the type.
+
+It's worth noting that there are no unsigned versions of numeric types in Substrait but they can be approximated with
+casts and user-defined types.
+
+| Type                 | Description                                                              |
+|----------------------|--------------------------------------------------------------------------|
+| bool                 | boolean                                                                  |
+|                      |                                                                          |
+| i8                   | one byte signed integer                                                  |
+| i16                  | two byte signed integer                                                  |
+| i32                  | four byte signed integer                                                 |
+| i64                  | eight byte signed integer                                                |
+|                      |                                                                          |
+| fp32                 | four byte signed float                                                   |
+| fp64                 | eight byte signed float                                                  |
+|                      |                                                                          |
+| string               | string                                                                   |
+|                      |                                                                          |
+| timestamp            | timestamp without a timezone                                             |
+| date                 | date (day resolution)                                                    |
+| time                 | time within a day (microsecond resolution)                               |
+|                      |                                                                          |
+| interval_year        | duration specified in years and months                                   |
+| interval_day         | duration specified in days, hours, minutes, seconds, and microseconds    |
+| timestamp_tz         | timestamp with a specified timezone                                      |
+| uuid                 | 128-bit UUID                                                             |
+|                      |                                                                          |
+| varchar\<L>          | a variable sized character sequence of up to length L                    |
```

Comment:
Why `\<`? Is this a markdown escape of some kind?

Reply:
Ok.  It appears `\` escapes are valid in common-mark (even though inline HTML is not) so this should be ok.